src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx

```diff
@@ -207,54 +215,60 @@ function EnvironmentSelectorComponent({isExternalApp, _init}: {isExternalApp: bo
                     )}
                     {appDetails?.deploymentAppDeleteRequest && (
                         <>
-                            <BinWithDots className='icon-dim-16 mr-8 ml-12'/>
+                            <BinWithDots className="icon-dim-16 mr-8 ml-12" />
                             <span className="cr-5 fw-6">{DELETE_DEPLOYMENT_PIPELINE}</span>
                             <span className="dc__loading-dots cr-5" />
                         </>
                     )}
                 </div>
             </div>
 
-            <div className="flex">
-                {!appDetails.deploymentAppDeleteRequest && (
-                    <button className="flex left small cta cancel pb-6 pt-6 pl-12 pr-12 en-2" onClick={showInfoUrl}>
-                        <LinkIcon className="icon-dim-16 mr-6 icon-color-n7" />
-                        Urls
-                    </button>
-                )}
-                {!showWorkloadsModal && (
-                    <button
-                        className="scale-workload__btn flex left cta cancel pb-6 pt-6 pl-12 pr-12 en-2 ml-6"
-                        onClick={() => setWorkloadsModal(true)}
-                    >
-                        <ScaleObjects className="mr-4" /> Scale workloads
-                    </button>    
-                )}
+            {!loadingResourceTree && (
+                <div className="flex">
+                    {!appDetails.deploymentAppDeleteRequest && (
+                        <button className="flex left small cta cancel pb-6 pt-6 pl-12 pr-12 en-2" onClick={showInfoUrl}>
+                            <LinkIcon className="icon-dim-16 mr-6 icon-color-n7" />
+                            Urls
+                        </button>
+                    )}
+                    {!showWorkloadsModal && (
+                        <button
+                            className="scale-workload__btn flex left cta cancel pb-6 pt-6 pl-12 pr-12 en-2 ml-6"
+                            onClick={() => setWorkloadsModal(true)}
```

Comment:
Avoid inline function

Reply:
Existing. We'll be covering this in upcoming PR..